monitoring_service/state_db/sqlite_db.py

```diff
@@ -35,22 +29,72 @@ def setup_db(self, network_id: int, contract_address: str, receiver: str):
         self.conn.execute(UPDATE_METADATA_SQL, [network_id, contract_address, receiver])
         self.conn.commit()
 
-    @property
-    def monitor_requests(self) -> dict:
-        c = self.conn.cursor()
-        c.execute('SELECT * FROM `monitor_requests`')
+    def is_initialized(self) -> bool:
+        self.cursor.execute(
+            "SELECT name FROM sqlite_master WHERE type='table' AND name='metadata'",
+        )
+        return self.cursor.fetchone() is not None
+
+    def fetch_scalar(self, query: str, query_args: Iterable = ()):
+        """ Helper function to fetch a single field of a single row """
+        self.cursor.execute(query, query_args)
+        return self.cursor.fetchone()[0]
+
+    def get_monitor_request_rows(
+        self,
+        channel_identifier: ChannelIdentifier = None,
+        non_closing_signer: Address = None,
+    ) -> List[dict]:
+        """ Fetch MRs form the db, optionally filtered """
+        query = 'SELECT * FROM monitor_requests WHERE 1=1'  # 1=1 for easier query building
+        query_args = []
+        if channel_identifier:
+            query += ' AND channel_identifier = ?'
+            query_args.append(hex(channel_identifier))
+        if non_closing_signer:
+            query += ' AND non_closing_signer = ?'
+            query_args.append(non_closing_signer)
+
+        self.cursor.execute(query, query_args)
         ret = []
-        for x in c.fetchall():
+        for x in self.cursor:
+            x = dict(x)
```

Comment:
Yes, it can (behaves both like dict and tuple). The `Row` is immutable, but we want to unhex some keys, so I used `dict()` to make it mutable for this case. I still plan to do this in a nicer way in a future PR.